### PR TITLE
rel_7_4 Merge-back

### PR DESCRIPTION
**Pull request by volodymyr-korzh**
Merge fixes from rel_7_4

# Resolved Conflicts:

- #    hapi-deployable-pom/pom.xml
- #    hapi-fhir-android/pom.xml
- #    hapi-fhir-base/pom.xml
- #    hapi-fhir-base/src/main/java/ca/uhn/fhir/repository/Repository.java
- #    hapi-fhir-bom/pom.xml
- #    hapi-fhir-checkstyle/pom.xml
- #    hapi-fhir-cli/hapi-fhir-cli-api/pom.xml
- #    hapi-fhir-cli/hapi-fhir-cli-app/pom.xml
- #    hapi-fhir-cli/pom.xml
- #    hapi-fhir-client-okhttp/pom.xml
- #    hapi-fhir-client/pom.xml
- #    hapi-fhir-converter/pom.xml
- #    hapi-fhir-dist/pom.xml
- #    hapi-fhir-docs/pom.xml
- #    hapi-fhir-jacoco/pom.xml
- #    hapi-fhir-jaxrsserver-base/pom.xml
- #    hapi-fhir-jpa/pom.xml
- #    hapi-fhir-jpaserver-base/pom.xml
- #    hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java
- #    hapi-fhir-jpaserver-elastic-test-utilities/pom.xml
- #    hapi-fhir-jpaserver-hfql/pom.xml
- #    hapi-fhir-jpaserver-ips/pom.xml
- #    hapi-fhir-jpaserver-mdm/pom.xml
- #    hapi-fhir-jpaserver-model/pom.xml
- #    hapi-fhir-jpaserver-searchparam/pom.xml
- #    hapi-fhir-jpaserver-subscription/pom.xml
- #    hapi-fhir-jpaserver-test-dstu2/pom.xml
- #    hapi-fhir-jpaserver-test-dstu3/pom.xml
- #    hapi-fhir-jpaserver-test-r4/pom.xml
- #    hapi-fhir-jpaserver-test-r4b/pom.xml
- #    hapi-fhir-jpaserver-test-r5/pom.xml
- #    hapi-fhir-jpaserver-test-utilities/pom.xml
- #    hapi-fhir-jpaserver-uhnfhirtest/pom.xml
- #    hapi-fhir-server-cds-hooks/pom.xml
- #    hapi-fhir-server-mdm/pom.xml
- #    hapi-fhir-server-openapi/pom.xml
- #    hapi-fhir-server/pom.xml
- #    hapi-fhir-serviceloaders/hapi-fhir-caching-api/pom.xml
- #    hapi-fhir-serviceloaders/hapi-fhir-caching-caffeine/pom.xml
- #    hapi-fhir-serviceloaders/hapi-fhir-caching-guava/pom.xml
- #    hapi-fhir-serviceloaders/hapi-fhir-caching-testing/pom.xml
- #    hapi-fhir-serviceloaders/pom.xml
- #    hapi-fhir-spring-boot/hapi-fhir-spring-boot-autoconfigure/pom.xml
- #    hapi-fhir-spring-boot/hapi-fhir-spring-boot-samples/hapi-fhir-spring-boot-sample-client-apache/pom.xml
- #    hapi-fhir-spring-boot/hapi-fhir-spring-boot-samples/hapi-fhir-spring-boot-sample-client-okhttp/pom.xml
- #    hapi-fhir-spring-boot/hapi-fhir-spring-boot-samples/hapi-fhir-spring-boot-sample-server-jersey/pom.xml
- #    hapi-fhir-spring-boot/hapi-fhir-spring-boot-samples/pom.xml
- #    hapi-fhir-spring-boot/hapi-fhir-spring-boot-starter/pom.xml
- #    hapi-fhir-spring-boot/pom.xml
- #    hapi-fhir-sql-migrate/pom.xml
- #    hapi-fhir-storage-batch2-jobs/pom.xml
- #    hapi-fhir-storage-batch2-test-utilities/pom.xml
- #    hapi-fhir-storage-batch2/pom.xml
- #    hapi-fhir-storage-cr/pom.xml
- #    hapi-fhir-storage-mdm/pom.xml
- #    hapi-fhir-storage-test-utilities/pom.xml
- #    hapi-fhir-storage/pom.xml
- #    hapi-fhir-structures-dstu2.1/pom.xml
- #    hapi-fhir-structures-dstu2/pom.xml
- #    hapi-fhir-structures-dstu3/pom.xml
- #    hapi-fhir-structures-hl7org-dstu2/pom.xml
- #    hapi-fhir-structures-r4/pom.xml
- #    hapi-fhir-structures-r4b/pom.xml
- #    hapi-fhir-structures-r5/pom.xml
- #    hapi-fhir-test-utilities/pom.xml
- #    hapi-fhir-testpage-overlay/pom.xml
- #    hapi-fhir-validation-resources-dstu2.1/pom.xml
- #    hapi-fhir-validation-resources-dstu2/pom.xml
- #    hapi-fhir-validation-resources-dstu3/pom.xml
- #    hapi-fhir-validation-resources-r4/pom.xml
- #    hapi-fhir-validation-resources-r4b/pom.xml
- #    hapi-fhir-validation-resources-r5/pom.xml
- #    hapi-fhir-validation/pom.xml
- #    hapi-tinder-plugin/pom.xml
- #    hapi-tinder-test/pom.xml
- #    pom.xml
- #    tests/hapi-fhir-base-test-jaxrsserver-kotlin/pom.xml
- #    tests/hapi-fhir-base-test-mindeps-client/pom.xml
- #    tests/hapi-fhir-base-test-mindeps-server/pom.xml

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java [hunk 1]
```diff
@@ -470,10 +471,26 @@ protected boolean shouldSkipTask() {
 			}
 		}
 
-		version.onTable(Search.HFJ_SEARCH)
-				.modifyColumn("20240722.1", Search.SEARCH_UUID)
-				.nonNullable()
-				.withType(ColumnTypeEnum.STRING, 48);
+		{
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java:474]: comment: I'm only reviewing the conflicts.  I'm assuming the rest merged cleanly

File: pom.xml [hunk 2]
```diff
@@ -1144,7 +1144,7 @@
 			<dependency>
 				<groupId>com.graphql-java</groupId>
 				<artifactId>graphql-java</artifactId>
-				<version>21.0</version>
+				<version>21.5</version>
```
Comment (michaelabuckley) [inline pom.xml:1147]: suggestion: I suspect this PR includes a breaking change.  Maybe bump the pom snapshot versions to be safe?  Unless you see CDR build clean without referencing this branch.

**Review by michaelabuckley — APPROVED** [review]
suggestion: I suspect this PR includes a breaking change. Maybe bump the pom snapshot versions to be safe? Unless you see CDR build clean without referencing this branch.

File: pom.xml (hunk 2 above)
Comment (volodymyr-korzh) [reply pom.xml:1147]: CDR build (compilation) succeeded for current CDR master,  there are several test failures, but looks like we can proceed without version bump

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java (hunk 1 above)
Comment (volodymyr-korzh) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java:474]: yes, there were only few conflicts that were easy to resolve, rest of the changes merged cleanly

**Review by tadgh — APPROVED** [review]

**Pull request merged**